src/interactivity/base/ServiceLocator.cpp

```diff
@@ -104,6 +104,27 @@ void ServiceLocator::RundownAndExit(const HRESULT hr)
 
 #pragma region Set Methods
 
+[[nodiscard]] NTSTATUS ServiceLocator::SetConsoleControlInstance(_In_ std::unique_ptr<IConsoleControl>& control)

```

Comment:
Augh, forcing ownership transfer like this is nonidiomatic.

If you want to codify it _in a C++ way_ you should take `std::unique_ptr<IConsoleControl> control` and pass it in with `std::move` and _move it out into `s_consoleControl` with `std::move` as well_.

Reply:
Even better:

```suggestion
[[nodiscard]] NTSTATUS ServiceLocator::SetConsoleControlInstance(_In_ std::unique_ptr<IConsoleControl>&& control)
```

this forces the caller to 🎵 `move` it `move` it